test/ziggurat/messaging/producer_test.clj

```diff
@@ -41,6 +41,57 @@
             message {:foo "bar" :retry-count @retry-count}
             expected-message {:foo "bar"}]
         (producer/retry-for-channel message topic-entity channel)
+        (while (> @retry-count 0)
+          (swap! retry-count dec)
+          (let [message-from-mq (rmq/get-message-from-channel-delay-queue topic-entity channel)]
+            (producer/retry-for-channel message-from-mq topic-entity channel)))
+        (let [message-from-mq (rmq/get-msg-from-channel-dead-queue topic-entity channel)]
+          (is (= expected-message message-from-mq))))))
+
+  (testing "message in channel will be retried with linear queue timeout"
+    (fix/with-queues
+      {:default {:handler-fn #(constantly nil)
+                 :linear-retry #(constantly nil)}}
+      (let [retry-count (atom 2)
+            topic-entity :default
+            channel :linear-retry
+            message {:foo "bar" :retry-count @retry-count}
+            expected-message {:foo "bar"}]
+        (producer/retry-for-channel message topic-entity channel)
+        (while (> @retry-count 0)
+          (swap! retry-count dec)
+          (let [message-from-mq (rmq/get-message-from-channel-delay-queue topic-entity channel)]
+            (producer/retry-for-channel message-from-mq topic-entity channel)))
+        (let [message-from-mq (rmq/get-msg-from-channel-dead-queue topic-entity channel)]
+          (is (= expected-message message-from-mq))))))
+
+  (testing "message in channel will be retried with exponential queue timeout"
```

Comment:
@thediantoni  how are we testing that exponential back-off was used? We're not measuring time taken anywhere.